### PR TITLE
Update VelociraptorLoader based on version 0.7.0

### PR DESCRIPTION
**Pull request by Zawadidone**
Add new Velociraptor NTFS accessors

### CODE REVIEW

**Comment by Zawadidone** [comment]
The volumes and paths are mapped correctly but the NTFS filesystem is not correctly parsed. I will try to find the bug that causes this.

```
Traceback (most recent call last):
  File "/dissect.target/dissect/target/target.py", line 404, in _load
    ldr.map(target)
  File "/dissect.target/dissect/target/loaders/velociraptor.py", line 75, in map
    map_dirs(
  File "/dissect.target/dissect/target/loaders/dir.py", line 54, in map_dirs
    loaderutil.add_virtual_ntfs_filesystem(target, dfs, **kwargs)
  File "/dissect.target/dissect/target/helpers/loaderutil.py", line 36, in add_virtual_ntfs_filesystem
    ntfs = NtfsFilesystem(boot=fh_boot, mft=fh_mft, usnjrnl=fh_usnjrnl, sds=fh_sds)
  File "/dissect.target/dissect/target/filesystems/ntfs.py", line 36, in __init__
    self.ntfs = NTFS(fh, boot=boot, mft=mft, usnjrnl=usnjrnl, sds=sds)
  File "/dissect.target/.tox/py3/lib/python3.10/site-packages/dissect/ntfs/ntfs.py", line 50, in __init__
    self.boot_sector = c_ntfs.BOOT_SECTOR(boot_fh)
  File "/dissect.target/.tox/py3/lib/python3.10/site-packages/dissect/cstruct/types/base.py", line 24, in __call__
    return self.read(*args, **kwargs)
  File "/dissect.target/.tox/py3/lib/python3.10/site-packages/dissect/cstruct/types/base.py", line 73, in read
    return self._read(obj)
  File "<compiled _BOOT_SECTOR>", line 15, in _read
EOFError
```

**Comment by Schamper** [comment]
Looks like it tries to open the `$Boot` file (which I'm not sure if Velociraptor collects?). It seems to exist (otherwise `boot_fh` would be `None`) but it's too small to actually contain a boot sector. If this is inherently a Velociraptor problem, we could put a try/except in the NTFS code that reads the boot sector.

**Comment by Zawadidone** [comment]
It does collect the file but with a file size of `0` which normally is not a problem, but the bug (https://github.com/Velocidex/velociraptor/issues/2892) has reintroduced itself for the second time. I don't understand the NTFS code good enough to determine if a try/except improves the loader, but if does could you add it to the PR or https://github.com/fox-it/dissect.ntfs.

I expect that the bug will be fixed in the next release of Velociraptor. I have changed the PR to ready for review because the loader functions correctly.

**Comment by Schamper** [comment]
A cleaner fix might be to check for `st_size > 0` in here: https://github.com/fox-it/dissect.target/blob/main/dissect/target/helpers/loaderutil.py#L44-L48. All of the files that need to be opened should be larger anyway.

File: tests/test_loaders_velociraptor.py [hunk 1]
```diff
@@ -50,7 +50,7 @@
         ),
     ],
 )
-def test_velociraptor_loader_windows_ntfs(paths: List[str], mock_target: Target, tmp_path: Path):
+def test_velociraptor_loader_windows_ntfs(paths: List[str], mock_target: Target, tmp_path: Path) -> None:
```
Comment (Schamper) [inline tests/test_loaders_velociraptor.py:53]:
```suggestion
def test_velociraptor_loader_windows_ntfs(paths: list[str], mock_target: Target, tmp_path: Path) -> None:
```

File: tests/test_loaders_velociraptor.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ def test_velociraptor_loader_windows_ntfs(paths: List[str], mock_target: Target,
         (["uploads.json", "uploads/auto/Library", "uploads/auto/Applications"]),
     ],
 )
-def test_dir_loader_unix(paths: List[str], mock_target: Target, tmp_path: Path):
+def test_dir_loader_unix(paths: List[str], mock_target: Target, tmp_path: Path) -> None:
```
Comment (Schamper) [inline tests/test_loaders_velociraptor.py:91]:
```suggestion
def test_dir_loader_unix(paths: list[str], mock_target: Target, tmp_path: Path) -> None:
```
Some cleanup of old style type hints.

File: dissect/target/helpers/loaderutil.py [hunk 3]
```diff
@@ -38,12 +41,17 @@ def add_virtual_ntfs_filesystem(
...
 
     for path in paths:
         try:
-            return fs.open(path)
+            path = fs.get(path)
+            if path.stat().st_size > 0:
+                return path.open()
+            else:
+                log.warning("The file size of %s is zero and thus cannot be parsed", path)
+                pass
```
Comment (Schamper) [inline dissect/target/helpers/loaderutil.py:54]:
```suggestion
            entry = fs.get(path)
            if entry.stat().st_size > 0:
                return path.open()
            else:
                log.warning("File is empty and will be skipped: %s", path)
```

File: dissect/target/helpers/loaderutil.py [hunk 4]
```diff
@@ -38,12 +41,17 @@ def add_virtual_ntfs_filesystem(
         fs.ntfs = ntfs.ntfs
 
 
-def _try_open(fs, path):
+def _try_open(fs, path) -> None:
```
Comment (Schamper) [inline dissect/target/helpers/loaderutil.py:44]:
```suggestion
def _try_open(fs: Filesystem, path: str) -> None:
```
And add the import for `from dissect.target.filesystem import Filesystem`

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/helpers/loaderutil.py [hunk 5]
```diff
@@ -1,12 +1,16 @@
+import logging
 import re
 import urllib
 from os import PathLike
 from pathlib import Path
 from typing import Optional, Tuple, Union
```
Comment (Schamper) [inline dissect/target/helpers/loaderutil.py:6]: While you're here, could you replace the type hint that currently uses `Tuple` with `tuple`? Thanks!

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/helpers/loaderutil.py [hunk 6]
```diff
@@ -38,17 +42,22 @@ def add_virtual_ntfs_filesystem(
         fs.ntfs = ntfs.ntfs
 
 
-def _try_open(fs, path):
+def _try_open(fs: Filesystem, path: str) -> None:
```
Comment (Schamper) [inline dissect/target/helpers/loaderutil.py:45]:
```suggestion
def _try_open(fs: Filesystem, path: str) -> BinaryIO:
```

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/helpers/loaderutil.py [hunk 7]
```diff
@@ -38,17 +42,22 @@ def add_virtual_ntfs_filesystem(
...
 
     for path in paths:
         try:
-            return fs.open(path)
+            path = fs.get(path)
+            if path.stat().st_size > 0:
+                return path.open()
+            else:
+                log.warning("File is empty and will be skipped: %s", path)
+                pass
```
Comment (Schamper) [inline dissect/target/helpers/loaderutil.py:55]:
```suggestion
```

**Review by Schamper — CHANGES_REQUESTED** [review]
Small nitpicky comment, good to go otherwise.

**Review by Schamper — APPROVED** [review]

**Pull request merged**